compiler/rustc_smir/src/rustc_smir/mod.rs

```diff
@@ -42,6 +42,17 @@ impl<'tcx> Context for Tables<'tcx> {
         Some(self.crate_item(self.tcx.entry_fn(())?.0))
     }
 
+    fn all_trait_decls(&mut self) -> stable_mir::TraitDecls {
+        self.tcx
+            .traits(LOCAL_CRATE)
+            .iter()
+            .map(|trait_def_id| {
+                let trait_def = self.trait_def(*trait_def_id);
+                self.trait_decl(&trait_def)
```

Comment:
Instead of eagerly converting them, you could just return the `TraitDef`s and leave it to the user to convert the individual items lazily.

Reply:
Makes sense.